### PR TITLE
Security fix for Prototype Pollution

### PR DESCRIPTION
**Pull request by gkmr507**
### 📊 Metadata *

`@blakek/deep` is vulnerable to `Prototype Pollution`.
This package allowing for modification of prototype behavior, which may result in Information Disclosure/DoS/RCE.

### ⚙️ Description *

Prototype Pollution refers to the ability to inject properties into existing JavaScript language construct prototypes, such as objects.
JavaScript allows all Object attributes to be altered, including their magical attributes such as proto, constructor and prototype.
An attacker manipulates these attributes to overwrite, or pollute, a JavaScript application object prototype of the base object by injecting other values.
Properties on the Object.prototype are then inherited by all the JavaScript objects through the prototype chain.

### 💻 Technical Description *

Fixed by avoiding setting magical attributes.

### 🐛 Proof of Concept (PoC) *

1. Create the following PoC file:

```js
// poc.js
var deep = require("@blakek/deep")
var obj = {}
console.log("Before : " + {}.polluted);
deep.set('Yes! Its Polluted', '__proto__.polluted', obj);
console.log("After : " + {}.polluted);
```

2. Execute the following commands in another terminal:

```bash
npm i @blakek/deep # Install affected module
node poc.js #  Run the PoC
```

3. Check the Output:
```
Before : undefined
After : Yes! Its Polluted
```

### 🔥 Proof of Fix (PoF) *

After fix execution will block prototype pollution and polluted will be [undefined.]

### 👍 User Acceptance Testing (UAT)

After fix functionality is unaffected.

### CODE REVIEW

**Review by mzfr — COMMENTED** [review]
LGTM!

P.S:There is already #1

File: src/index.ts [hunk 1]
```diff
@@ -74,6 +74,7 @@ function _set(value: any, path: Path, object: any): any {
   let reference = object;
 
   parsedPath.forEach((key, index) => {
+    if (isPrototypePolluted(key)) return false;
```
Comment (Mik317) [inline src/index.ts:77]: Better returning a `unmodified object` since the method returns a `object`

**Review by Mik317 — CHANGES_REQUESTED** [review]
Returning a different type from the one the `method` needs could make problems or errors in code where the library was already in use.

EDIT: I noted the `return` the line after. Any thought on what's the better implementation?

Cheers,
Mik

**Comment by huntr-helper** [comment]
Sorry hethvik, we enjoyed reviewing your fix but it has not been selected this time. If this bounty has not been closed, please feel free to try again with a new pull request!

We appreciate your effort and look forward to reviewing more of your fixes in the future! 🔨😎

[Come join us on Discord](https://huntr.dev/discord)

**Pull request closed without merging**